networkx/algorithms/dag.py

```diff
@@ -287,3 +289,75 @@ def is_aperiodic(G):
         return g==1
     else:
         return g==1 and nx.is_aperiodic(G.subgraph(set(G)-set(levels)))
+
+def transitive_closure(G):
+    """ Based on http://www.ics.uci.edu/~eppstein/PADS/PartialOrder.py
+
+    The transitive closure of G = (V,E) is a graph G+ = (V,E+) such that
+    for all v,w in V there is an edge (v,w) in E+ if and only if there
+    is a non-null path from v to w in G.
+
+    Parameters
+    ----------
+    G : NetworkX DiGraph
+      Graph
+
+    Returns
+    -------
+    TC : NetworkX DiGraph
+       Graph
+
+    Raises
+    ------
+    NetworkXError
+      If G is not directed
+
+    """
+    if not G.is_directed():
+        raise nx.NetworkXError("transitive_closure not defined for undirected graphs")
+    TC = nx.DiGraph()
+    TC.add_nodes_from(G.nodes_iter())
+    TC.add_edges_from(G.edges_iter())
+    for v in G:
+        TC.add_edges_from((v, u) for u in nx.dfs_preorder_nodes(G, source=v)
+                          if v != u)
+    return TC
+
+def antichain_generator(G):
+    """Generates antichains from a DAG.
+
```

Comment:
It would be better to find a definition from an authoritative source (eg a scientific paper or book, does Wikipedia cite any interesting source?). But Wikipedia is OK if we do not find anything else.